libs/sync/tables.js

```diff
@@ -161,11 +168,8 @@ const syncColumns = (api, apps, opts) => {
         }, Promise.resolve())
 }
 
-
 module.exports = (api, apps, opts) =>
-    Promise.resolve()
-        .then(() => syncTables(api, apps, opts))
-        // update table data
-        .then(() => api.getAppDataTables())
-        .then(() => syncColumns(api, apps, opts))
-        .then(() => cleanup(api, apps))
\ No newline at end of file
+   syncTables(api, apps, opts)
+      .then(() => api.getAppDataTables(opts.columnsToIgnore))
+      .then(() => syncColumns(api, apps, opts))
+      .then(() => cleanup(api, apps))
```

Comment:
If you decided to refactor it, why not replace all these constructions with async/await?

```suggestion
  await syncTables(api, apps, opts)
  await api.getAppDataTables(opts.columnsToIgnore)
  await syncColumns(api, apps, opts)
  await cleanup(api, apps)
```